builder/amazon/common/run_config.go

```diff
@@ -83,5 +85,12 @@ func (c *RunConfig) Prepare(ctx *interpolate.Context) []error {
 		}
 	}
 
+	m, _ := regexp.MatchString("(stop|terminate)", c.InstanceInitiatedShutdownBehavior)
```

Comment:
This regexp should be declared in package scope; e.g.

```
var reShutdownBehavior = regexp.MustCompile("(stop|terminate)")
```

This way it will be checked at compile time instead of runtime and we don't have an unchecked error here.